steps/math/batch_aggregate.go

```diff
@@ -0,0 +1,109 @@
+package math
+
+import (
+	"fmt"
+	"github.com/bitflow-stream/go-bitflow/bitflow"
+	"github.com/bitflow-stream/go-bitflow/script/reg"
+)
+
+type Aggregator interface {
+	Aggregate(header *bitflow.Header, samples []*bitflow.Sample, refSample *bitflow.Sample) (*bitflow.Sample, error)
+	String() string
+}
+
+type SumAggregator struct{}
+
+func (a *SumAggregator) Aggregate(header *bitflow.Header, samples []*bitflow.Sample, refSample *bitflow.Sample) (*bitflow.Sample, error){
+	values := make([]bitflow.Value, len(header.Fields))
+	for _, sample := range samples{
+		for i, value := range sample.Values {
+			values[i] += value
+		}
+	}
+	sample := refSample.Clone()
+	sample.Values = values
+	return sample, nil
+}
+
+func (a *SumAggregator) String() string {
+	return "Sum Aggregator"
+}
+
+type MultiplyAggregator struct{}
+
+func (a *MultiplyAggregator) Aggregate(header *bitflow.Header, samples []*bitflow.Sample, refSample *bitflow.Sample) (*bitflow.Sample, error){
+	values := make([]bitflow.Value, len(header.Fields))
+	for i, sample := range samples{
+		for j, value := range sample.Values {
+			if i ==  0 {
+				values[j] = value
+			} else {
+				values[j] *= value
+			}
+		}
+	}
+	sample := refSample.Clone()
+	sample.Values = values
+	return sample, nil
+}
+
+func (a *MultiplyAggregator) String() string {
+	return "Multiply Aggregator"
+}
+
+type AverageAggregator struct{}
+
+func (a *AverageAggregator) Aggregate(header *bitflow.Header, samples []*bitflow.Sample, refSample *bitflow.Sample) (*bitflow.Sample, error){
+	values := make([]bitflow.Value, len(header.Fields))
+	for _, sample := range samples{
+		for i, value := range sample.Values {
+			values[i] += value
+		}
+	}
+	for i, _ := range values {
+		values[i] /= bitflow.Value(len(samples))
+	}
+	sample := refSample.Clone()
+	sample.Values = values
+	return sample, nil
+}
+
+func (a *AverageAggregator) String() string {
+	return "Average Aggregator"
+}
+
+type BatchAggregator struct {
+	bitflow.NoopProcessor
+
+	aggregator Aggregator
+}
+
+func (ba *BatchAggregator) ProcessBatch(header *bitflow.Header, samples []*bitflow.Sample) (*bitflow.Header, []*bitflow.Sample, error) {
+	samples = make([]*bitflow.Sample, 1)
+	samples[0], _= ba.aggregator.Aggregate(header, samples, samples[len(samples) -1])
```

Comment:
If the error return parameter is ignored, it should not be defined in the interface. Either return the error here (if there is one), or remove the option of returning an error from Aggregate(). I would prefer the second option, because I think that a simple aggregation operation should never fail. Correct me if I'm wrong?